Aplib.Core/Belief/Belief.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+
+namespace Aplib.Core.Belief
+{
+    /// <summary>
+    /// The <see cref="Belief{TReference, TResource}"/> class represents a belief.
+    /// Some <i>object reference</i> is used to generate/update a <i>resource</i> 
+    /// (i.e., some piece of information on the game state as perceived by an agent).
+    /// </summary>
+    /// <remarks>
+    /// It implements the <see cref="IBelief"/> interface.
+    /// It supports implicit conversion to <typeparamref name="TResource"/>.
+    /// </remarks>
+    /// <typeparam name="TReference">The type of the reference used to generate/update the resource.</typeparam>
+    /// <typeparam name="TResource">The type of the resource the belief represents.</typeparam>
+    public class Belief<TReference, TResource> : IBelief
+    {
+        /// <summary>
+        /// The reference used to generate/update the resource.
+        /// </summary>
+        private readonly TReference _reference;
```

Comment:
We discussed this IRL and concluded that this would be most clear. It's not a reference to the observation per se. It is _any_ object reference that can be used to update the observation. Consider, for example, an enemy position as an observation; the reference would not be an object reference to the _enemy position_, as this is a value type and not a reference type. The reference would be the _enemy transform_. `TObservationReference _observationReference` would thus be a misleading name. I think `TReference _reference` is still the best name.